pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeUtils.java

```diff
@@ -119,7 +119,8 @@
     public static List<String> getGoInstanceCmd(InstanceConfig instanceConfig,
                                                 String originalCodeFileName,
                                                 String pulsarServiceUrl,
-                                                boolean k8sRuntime) throws IOException {
+                                                boolean k8sRuntime,
+                                                int metricsPort) throws IOException {
```

Comment:
Why not introduce the new configuration in the `InstanceConfig`?

Reply:
I havnt add `metricsPort` to `InstanceConfig` is because `metricsPort` is generated with different mechanisms for different runtime, such as for process runtime, `metricsPort` is generated by `FunctionCommon.findAvailablePort()` but for k8s runtime, `metricsPort` comes from the config `functions_worker.yml`.

Also, for java and python function, `metricsPort` is also not included in `InstanceConfig` (ref: https://github.com/apache/pulsar/blob/7c68ade453136895d19fa4723351d26256f3d624/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeUtils.java#L249).

I did seem a bit lazy with these changes here, but also want to avoid making changes to java and python runtime's code. So let me know if we want apply `metricsPort` into `InstanceConfig`.